### PR TITLE
Migrate Endpoints v2 template to OSS vLLM

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
https://github.com/anyscale/ray-llm/pull/1883 made RayLLM images use OSS vLLM by default. This change updates the Endpoints v2 template to use OSS vLLM fields.

### CODE REVIEW

File: templates/endpoints_v2/examples/CustomModels.ipynb [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "\n",
     "The string template should include the `{instruction}` keyword, which will be replaced with message content from the ChatCompletions API.\n",
     "\n",
-    "For example, if a user sends the following message for llama2-7b-chat-hf ([prompt format](models/llama/meta-llama--Llama-2-7b-chat-hf_a10g_tp1.yaml#L30-L36)):\n",
+    "For example, if a user sends the following message for `llama2-7b-chat-hf` ([prompt format](models/llama/meta-llama--Llama-2-7b-chat-hf_a10g_tp1.yaml#L30-L36)):\n",
```
Comment (akshay-anyscale) [inline templates/endpoints_v2/examples/CustomModels.ipynb:54]: oh this link may be wrong now. Can you check?

File: templates/endpoints_v2/examples/function_calling/DeployFunctionCalling.ipynb [hunk 2]
```diff
@@ -19,17 +19,15 @@
     "You can use the `generate_config.py` script to generate a model configuration YAML for function calling. \n",
     "\n",
     "These are the fields that will be set in the resulting model YAMLs for function calling -\n",
-    "1. `enable_json_logits_processors: true` under `engine_kwargs`\n",
+    "1. Environment variables under `runtime_env`\n",
```
Comment (akshay-anyscale) [inline templates/endpoints_v2/examples/function_calling/DeployFunctionCalling.ipynb:22]: is this really how we are enabling it? Definitely needs to be cleaned up - we should add this to list of tasks @shrekris-anyscale

**Review by comaniac — APPROVED** [review]
LGTM to the speculative decoding configurations.

File: templates/endpoints_v2/examples/function_calling/DeployFunctionCalling.ipynb (hunk 2 above)
Comment (shrekris-anyscale) [reply templates/endpoints_v2/examples/function_calling/DeployFunctionCalling.ipynb:22]: Yes it is. @kouroshHakha do we already have a plan to clean this up?

**Review by akshay-anyscale — APPROVED** [review]

File: templates/endpoints_v2/examples/CustomModels.ipynb (hunk 1 above)
Comment (shrekris-anyscale) [reply templates/endpoints_v2/examples/CustomModels.ipynb]: You're right. I update this section.

**Pull request merged**